cdap-docs/admin-manual/source/security/authorization.rst

```diff
@@ -354,97 +347,102 @@ Kerberos Principal
+    =====================  =============================================
+        Entity Type                 Entity Id
+    =====================  =============================================
+    ``namespace``          ``<namespace>:<namespace-id>``
+    ``application``        ``<namespace-id>.<app-id>``
+    ``program``            ``<namespace-id>.<app-id>.<program-type>.<program-id>``
+    ``dataset``            ``<namespace-id>.<dataset-id>``
+    ``stream``             ``<namespace-id>.<stream-id>``
+    ``artifact``           ``<namespace-id>.<artifact-id>``
+    ``dataset-type``       ``<namespace-id>.<dataset-type-id>``
+    ``dataset_module``     ``<namespace-id>.<dataset-module-id>``
+    ``securekey``          ``<namespace-id>.<secure-key-id>``
+    ``kerberosprincipal``  ``<kerberos-principal-id>``
+    =====================  =============================================
 
-- To add the role to other principal, use::
+  - For programs, the program type is one of:
+    ``flow``, ``mapreduce``, ``service``, ``spark``, ``worker``, or ``workflow``.
 
-    > add role <role-name> to <principal-type> <principal-name>
+  - Wildcard can be used in the entity name to grant privileges on multiple entities. For example,
 
-  where:
+    - ``namespace:ns*`` represents all namespaces that starts with ``ns``.
+    - ``namespace:ns?`` represents all namespaces that starts with ``ns`` and followed by a single character.
+    - ``program:ns1.app1.*`` represents all the programs in application ``app1`` in namespace ``ns1``.
 
-  - ``<role-name>`` is the role name that adds to the principal.
+- To add the role to a group, use::
 
-  - ``<principal-type>`` can **only** be ``group``.
-
-- To create a new role, use::
-
-    > create role <role-name>
+    > add role <role-name> to group <group-name>
 
 - To check the results, list the privileges for a principal::
 
     > list privileges for <principal-type> <principal-name>
 
-For example,
-to make ``alice``, in group ``admin``, as the administrator on a namespace ``ns1`` in a new environment,
-do the following steps:
+**Example**
+To give ``alice`` (who belongs to group ``admin``), *ADMIN* privilege on namespace ``ns1``, and all the
+entities in the namespace, do the following:
 
 - create a new role ``ns1_administrator``
 
-- use the commands to grant *ADMIN* on these entities: ``namespace:ns1``, ``application:ns1.*``, ``program:ns1.*.*``,
-  ``artifact:ns1.*``, ``dataset:ns1.*``, ``stream: ns1.*``, ``dataset_type:ns1.*``, ``dataset_module:ns1.*``,
-  ``securekey:ns1.*`` and ``kerberosprincipal.*`` to the role ``ns1_administrator``
+- grant the role ``ns1_administrator`` *ADMIN* on these entities:
 
-- add ``ns1_administrator`` to group ``admin``
+  - ``namespace:ns1``
+  - ``application:ns1.*``
+  - ``program:ns1.*.*``
+  - ``artifact:ns1.*``
+  - ``dataset:ns1.*``
+  - ``stream:ns1.*``
+  - ``dataset_type:ns1.*``
+  - ``dataset_module:ns1.*``
+  - ``securekey:ns1.*``
```

Comment:
We are missing `kerberosprincipal.*` here

Reply:
I removed it since kerberos principal is not part of a namespace

Reply:
I see.